### ISSUE TITLE
Crash when starting v0.30 due to breakage of window layout

### ISSUE THREAD

**Issue by baldurk** [baldurk/renderdoc#310]
### TL;DR - If RenderDoc v0.30 crashes on startup, delete your default window layout `%APPDATA%\renderdoc\DefaultLayout.config`

I'm not sure if this is new to v0.30 loading a layout saved on v0.29, or what, but I've seen a couple of people run into a problem with this callstack:

```
  at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at WeifenLuo.WinFormsUI.Docking.DockPanel.Persistor.LoadFromXml(DockPanel dockPanel, Stream stream, DeserializeDockContent deserializeContent, Boolean closeStream) in 3rdparty/WinFormsUI/Docking/DockPanel.Persistor.cs:line 625
   at WeifenLuo.WinFormsUI.Docking.DockPanel.Persistor.LoadFromXml(DockPanel dockPanel, String fileName, DeserializeDockContent deserializeContent) in 3rdparty/WinFormsUI/Docking/DockPanel.Persistor.cs:line 381
   at renderdocui.Windows.MainWindow.LoadLayout(Int32 layout) in MainWindow.cs:510
   at renderdocui.Windows.MainWindow.MainWindow_Load(Object sender, EventArgs e) in MainWindow.cs:174
```

which is right at the start. The direct cause of the crash is a 'pane' which is referencing a content that doesn't exist (either a negative ID, or much more likely a higher ID than contents that exist).

The strange thing is I can't see any obvious way that a layout could be corrupted like this, nor any change I've made recently (or ever) which could cause it - this is before it even calls in to persist string processing to do custom layout of individual windows.

If anyone runs into this, please send me your layout file so I can take a look at it!

**Comment by baldurk** [comment]
Someone passed me a layout file showing this bug happening.

It's caused by having multiple mesh viewers open - when loading on v0.30 only one gets created, which leads to an invalid index being used.

**Comment by baldurk** [comment]
This is fixed now, but I'm leaving this bug open so people will find it more easily and see the solution. I'll close it once I release v0.31.

**Comment by Marqin** [comment]
What's the path in Linux? I don't see dirs like`~/.renderdoc` nor `~/.config/renderdoc`.

**Comment by baldurk** [comment]
I don't follow. This bug only affects Windows because Linux basically doesn't have a UI yet. At the very least the non-functional UI doesn't have layouts. Are you seeing a crash?

**Comment by Marqin** [comment]
No I don't. But I wanted to post info on package page, in case someone got
it.

W dniu poniedziałek, 8 sierpnia 2016 Baldur Karlsson <
notifications@github.com> napisał(a):
